### ISSUE TITLE
keras accuracy goes down dramatically after loading saved model.

### ISSUE THREAD

**Issue by Ildar5** [keras-team/keras#14231]
`best_model.save('./best_model')`
`best_model.evaluate(test_images, test_labels)`

_INFO:tensorflow:Assets written to: ./best_model/assets
INFO:tensorflow:Assets written to: ./best_model/assets
12/12 [==============================] - 0s 9ms/step - loss: 0.7783 - accuracy: 0.8038
[0.7782523036003113, 0.8037634491920471]_

`loaded_model = keras.models.load_model('./best_model')`
`loaded_model.evaluate(test_images, test_labels)`

_12/12 [==============================] - 0s 9ms/step - loss: 0.7783 - accuracy: 0.1909
[0.7782523036003113, 0.19086021184921265]_

**Comment by Ildar5** [comment]
I did like so:

best_model.save('./best_model')
best_model.save_weights('./best_model_w')

best_model.evaluate(test_images, test_labels)
loss: 0.5851 - accuracy: 0.7930

and then

loaded_model = keras.models.load_model('./best_model')
loaded_model.load_weights('./best_model_w')

loaded_model.compile(
optimizer = 'adam',
loss = keras.losses.SparseCategoricalCrossentropy(),
metrics = ['accuracy']
)`

loaded_model.evaluate(test_images, test_labels)
loss: 0.5851 - accuracy: 0.7930

**Comment by adhilcodes** [comment]
where to fix this issue , I mean the where is the codebase of it.

**Comment by jvishnuvardhan** [comment]
@Ildar5 Sorry for the late response. Is this still an issue for you? 

Can you please share a simple standalone code to reproduce the issue? I noticed that similar issues were resolved in recent TF versions. Please check with recent version and also share a standalone code. Thanks!
